app/actions/websocket/posts.ts

```diff
@@ -0,0 +1,177 @@
+    // actions.push(receivedNewPost(post, collapsedThreadsEnabled));
+    if (shouldIgnorePost(post)) {
+        const {member} = await updateLastPostAt(serverUrl, post.channel_id, post.create_at, true);
+        if (member) {
+            models.push(member);
+        }
+    } else {
+        let markAsViewed = false;
+        let markAsRead = false;
+
+        if (!myChannel.manuallyUnread) {
+            if (
+                post.user_id === currentUserId &&
+                !isSystemMessage(post) &&
+                !isFromWebhook(post)
+            ) {
+                markAsViewed = true;
+                markAsRead = false;
+            } else if ((post.channel_id === currentChannelId)) { // TODO: THREADS && !viewingGlobalThreads) {
+                // Don't mark as read if we're in global threads screen
+                // the currentChannelId still refers to previously viewed channel
+                markAsViewed = false;
+                markAsRead = true;
+            }
+        }
+
+        if (markAsRead) {
+            markChannelAsRead(serverUrl, post.channel_id);
+            const {member} = await updateLastPostAt(serverUrl, post.channel_id, post.create_at, true);
+            if (member) {
+                models.push(member);
+            }
+        } else if (markAsViewed) {
+            const {member: viewedAt} = await markChannelAsViewed(serverUrl, post.channel_id, post.create_at, true);
+            if (viewedAt) {
+                models.push(viewedAt);
+            }
+        } else {
+            const hasMentions = msg.data.mentions.includes(currentUserId);
+            const {member: unreadAt} = await markChannelAsUnread(serverUrl, post.channel_id, myChannel.messageCount + 1, myChannel.mentionsCount + (hasMentions ? 1 : 0), false, post.create_at, undefined, true);
+            if (unreadAt) {
+                models.push(unreadAt);
+            }
+        }
+    }
+
+    database.operator.batchRecords(models);
+}
+
+export async function handlePostEdited(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const post: Post = JSON.parse(msg.data.post);
+
+    try {
+        fetchPostAuthors(serverUrl, [post]);
+    } catch {
+        // Do nothing
+    }
+
+    database.operator.handlePosts({
+        actionType: ActionType.POSTS.RECEIVED_NEW,
+        order: [post.id],
+        posts: [post],
+    });
+}
+
+export function handlePostDeleted(serverUrl: string, msg: WebSocketMessage) {
+    const data: Post = JSON.parse(msg.data.post);
+
+    markPostAsDeleted(serverUrl, data);
+}
+
+export async function handlePostUnread(serverUrl: string, msg: WebSocketMessage) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+
+    const postNumber = await database.database.get(POST).query(Q.where('channel_id', msg.broadcast.channel_id)).fetchCount();
```

Comment:
This seems error prone to say the least. If we move the msg count to a boolean, we can just pass this as true.

Reply:
this won't work as the amount of posts that you have loaded may very well not be the total amount of messages in the channel, we could resolve this by having the websocket to include the delta and for previous servers we could fetch the channel and use the value from the response.

Btw `gekidou` already has `is_unread` in the my_channel table.

Another alternative that I'm not fun of is to include `total_message_count` in the channel table, but definitely not a fan as this will cause the channel to update and re-render in a lot of places + the need to update it in lots of places manually and may cause out of sync data.

Reply:
I have gone the route of asking the server for the channel.